core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.coordinator.transaction
+
+import java.util
+
+import kafka.common.{InterBrokerSendThread, RequestAndCompletionHandler}
+import kafka.server.{DelayedOperationPurgatory, KafkaConfig, MetadataCache}
+import kafka.utils.Logging
+import org.apache.kafka.clients._
+import org.apache.kafka.common.metrics.Metrics
+import org.apache.kafka.common.network._
+import org.apache.kafka.common.requests.{TransactionResult, WriteTxnMarkersRequest}
+import org.apache.kafka.common.security.JaasContext
+import org.apache.kafka.common.utils.Time
+import org.apache.kafka.common.Node
+
+import java.util.concurrent.BlockingQueue
+
+import org.apache.kafka.common.protocol.Errors
+
+import collection.JavaConverters._
+
+case class CoordinatorEpochAndMarkers(metadataPartition: Int, coordinatorEpoch: Int, txnMarkerEntries: util.List[WriteTxnMarkersRequest.TxnMarkerEntry])
+case class DestinationBrokerAndQueuedMarkers(destBrokerNode: Node, markersQueue: BlockingQueue[CoordinatorEpochAndMarkers])
```

Comment:
I'm wondering if we should bound the size of this blocking queue or it will cause TC-broker OOM when some of the txn involved partition leaders are temporarily / permanently available. My feeling is that we do not and here is my reasoning (cc @junrao @dguy ):

1. We have two in-memory structures whose size is unbounded, this blocking queue and the txn purgatory.
2. When there are some partitions unavailable, hence the current prepareXX transaction cannot be completed, they will take one slot on each of the above structures;
3. Then when new request is coming from the same Pid, either `initPid` or `addPartitions`, they will not proceed. Hence the above structures will have at most one parked item for each producer.

See my other comment about handling the `initPid` and `addPartitions` request while the previous txn has not completed.